_delphi_utils_python/delphi_utils/flash_eval/eval_day.py

```diff
@@ -0,0 +1,332 @@
+    fips = pd.read_csv(f'{flash_dir}/geo_fips.csv', index_col=0).set_index('geo')
+    orig_fips = fips[fips['pop'].isna()].index
+    geos_repl = [fips_to_STATE[x] for x in orig_fips.str[:2]]
+    fips.loc[orig_fips, 'pop'] = [float(fips[fips.index == x]['pop']) for x in geos_repl]
+    fips_pop_table = fips.unstack().to_frame().T
+    fips_pop_table.columns = [STATE_to_fips[x] if x in list(STATES)
+                          else x for x in fips_pop_table.columns.droplevel()]
+    return STATE_to_fips, fips_pop_table
+def flash_eval(lag, day, input_df, signal, params, logger=None):
+    """Evaluate most recent data using FlaSH.
+
+    Input:
+    lag: the difference between the reporting and reference date
+    day: the day of the reference date (today is the reporting date)
+    input_df: a df from the day for a particular signal that includes natl. state, and county data
+    params: additional params needed.
+    Ouput:
+    None
+    """
+    print("RF")
+    if not logger:
+        logger = get_structured_logger(
+            name=signal,
+            filename=params["common"].get("log_filename", None),
+            log_exceptions=params["common"].get("log_exceptions", True))
+
+    #TODOv4: Change these to a local dir
+    flash_dir = f'flash_ref/{signal}'
+    last_7 = pd.read_csv(f'{flash_dir}/last_7_{lag}.csv', index_col=0).astype(float)
+    wk_mean = pd.read_csv(f'{flash_dir}/weekday_mean_df_{lag}.csv', index_col=0)
+    wk_var = pd.read_csv(f'{flash_dir}/weekday_var_df_{lag}.csv', index_col=0)
+    weekday_params = pd.read_csv(f'{flash_dir}/weekday_params_{lag}.csv', index_col=0)
+    summary_stats = pd.read_csv(f'{flash_dir}/summary_stats_{lag}.csv', index_col=0)
+    stream = pd.read_csv(f'{flash_dir}/ret_df2_{lag}.csv', index_col=0)
+
+
+    STATE_to_fips, fips_pop_table = setup_fips(flash_dir)
+    input_df.columns = [str(STATE_to_fips[x]) if x in list(STATES)
+                        else x for x in input_df.columns]
+
+    # discuss where to do out-of-range handling
+    out_range = input_df.columns[input_df.lt(int(params['flash']['support'][0])).iloc[0, :].values
+                        & input_df.gt(int(params['flash']['support'][1])).iloc[0, :].values ]
+
+
+    #only rank streams without out of range data
+    input_df = input_df[filter(lambda x: x not in out_range, input_df.columns)]
+
+    daily_update_df, non_daily_df_test, non_ar_df = split_reporting_schedule_dfs(input_df,
+                                                                                 flash_dir, lag)
+    # Weekday outlier [only for Daily Df]
+    weekday_outlier = daily_update_df.columns[((abs(
+        daily_update_df.T.sort_index()[day] - wk_mean.loc[day.day_of_week,
+        daily_update_df.columns].sort_index()) /wk_var.loc[day.day_of_week,
+        daily_update_df.columns].clip(1)).gt(5))]
+
+    # Make weekday correction for daily update
+    additive_factor = summary_stats[daily_update_df.columns].iloc[4, :]
```

Comment:
suggestion: Why does `iloc` use "4" here? Is that a special row? Recommend using a filter to make sure you're always getting the right row and so the code is easier to understand.

Reply:
Sorry, I'm going to replace the 2's and 4's - basically the columns refer to [25,50(median),75%] Quantiles and the mean and variance. I'm going to change them to words as per your recommendation!